packages/connect/src/base/stun.ts

```diff
@@ -300,44 +293,51 @@ function decodeIncomingSTUNResponses(addrs: Request[], socket: Socket, ms: numbe
   })
 }
 
-function getUsableResults(results: Request[], runningLocally = false): Request[] {
-  let filtered: Request[] = []
+/**
+ * Remove unusable responses from results
+ * @param responses results to filter
+ * @param runningLocally whether to run in local-mode or not
+ * @returns filtered results
+ */
+export function getUsableResults(responses: Request[], runningLocally = false): RequestWithResponse[] {
+  let filtered: RequestWithResponse[] = []
 
-  for (const result of results) {
-    if (!result.response) {
+  for (const result of responses) {
+    if (!isRequestWithResponse(result)) {
       continue
     }
 
-    switch (result.response.family) {
+    switch (result.response.family as 'IPv6' | 'IPv4') {
```

Comment:
We should rather put the `'IPv4' | 'IPv6'` type constraint on the `family` definition at https://github.com/hoprnet/hoprnet/blob/c866fd8f1009ec99666074db8ae758676c6434cf/packages/connect/src/base/stun.ts#L81 
We do that similarly for `Interface` type on line 14.

Reply:
Changed